### PR TITLE
*: Fix misuse of pkg/errors.Errorf and error directive

### PR DESCRIPTION
**Pull request by kakkoyun**
`%w` directive is only valid for recently introduces go native error wrapping. `pkg/errors` uses `%v`.
This PR fixes discovered issues in the code base.

Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Verification

* `make test-local`
* `make test-e2e`

### CODE REVIEW

File: pkg/replicate/replicater.go [hunk 1]
```diff
@@ -171,7 +171,7 @@ func RunReplicate(
 		level.Info(logger).Log("msg", "running replication attempt")
 
 		if err := newReplicationScheme(logger, metrics, blockFilter, fromBkt, toBkt, reg).execute(ctx); err != nil {
-			return errors.Errorf("replication execute: %w", err)
+			return errors.Errorf("replication execute: %v", err)
```
Comment (GiedriusS) [inline pkg/replicate/replicater.go]: Why don't we replace these with `errors.Wrap`?

File: pkg/prober/http_test.go [hunk 2]
```diff
@@ -72,7 +72,7 @@ func TestHTTPProberMuxRegistering(t *testing.T) {
 
 	var g run.Group
 	g.Add(func() error {
-		return errors.Errorf("serve probes %w", http.Serve(l, mux))
+		return errors.Errorf("serve probes %v", http.Serve(l, mux))
```
Comment (GiedriusS) [inline pkg/prober/http_test.go]: This seems like the only legit usage of `errors.Errorf` in these changed lines.

File: pkg/replicate/replicater.go (hunk 1 above)
Comment (kakkoyun) [reply pkg/replicate/replicater.go]: Actually, yes. This approach would be better. I haven't dug that into what they do while I was changing those. I'm gonna fix them.

Comment (bwplotka) [reply pkg/replicate/replicater.go]: Yes, Wrap would be better ((:

**Comment by kakkoyun** [comment]
@GiedriusS I've fixed misuse of `errors.Errorf` for the existing changes. Also traversed the code base and fixed other misuses that caught by my eyes.

I'd be glad if you could take another look at it.

**Pull request merged**